packages/expo-constants/src/ExponentConstants.web.ts

```diff
@@ -1,16 +1,21 @@
 import uuidv4 from 'uuid/v4';
 import UAParser from 'ua-parser-js';
+import { DeviceUUID } from 'device-uuid';
```

Comment:
Looking at this library, it seems like it attempts to fingerprint devices rather than generating a unique ID. Also if the device changes in a way that perturbs the ID generator (e.g. language settings), we'll return a different ID across page loads.

I think another approach that's truer to the spirit of the installation ID would be to generate a random UUID (uuidv4) and then store it in localStorage under a key like `expo.installationId`.

Reply:
From what I understand people use this value for analytic purposes. `installationId` get's aliased to `deviceId` which seems more fitting for what this prop is attempting. But I'm fine either way.

Reply:
`deviceId` on native is a misnomer and is different across apps and re-installations (for now). The way it works is we generate a random UUID and then save it in NSUserDefaults/SharedPreferences -- the web equivalent would be to write a value to localStorage / IndexedDB.